### PR TITLE
MEN-4079: dbus and mender clients

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -20,3 +20,13 @@ include:
...
+
+test:unit:
+  stage: test
+  script:
+    # start the dbus service
+    - service dbus start
+    # original from the gitlab-ci-check-golang-unittests.yml template
+    - go list ./... | grep -v vendor | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {} || exit $?
+    - mkdir -p tests/unit-coverage && find . -name 'coverage.txt' -exec cp --parents {} ./tests/unit-coverage \;
+    - tar -cvf ${CI_PROJECT_DIR}/unit-coverage.tar tests/unit-coverage
```
Comment (merlin-northern) [inline .gitlab-ci.yml:32]: so this is only until we update the template, right?

Comment (tranchitella) [reply .gitlab-ci.yml:32]: No idea, I'm asking @lluiscampos

**Review by merlin-northern — APPROVED** [review]
some of it should go into go-lib-micro but lets merge it, so we can continue.

File: client/mender/auth.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+// AuthClient is the interface for the Mender Authentication Manager clilents
+type AuthClient interface {
+	// Connect to the Mender client interface
+	Connect(objectName, objectPath, interfaceName string) error
+	// GetJWTToken returns a device JWT token
+	GetJWTToken() (string, error)
+	// FetchJWTToken schedules the fetching of a new device JWT token
+	FetchJWTToken() (bool, error)
```
Comment (merlin-northern) [inline client/mender/auth.go:36]: ok, so to get the token I call Get, so, when do I call Fetch?

Comment (tranchitella) [reply client/mender/auth.go:36]: Get returns you the current valid token (if any); fetch requires a new fetch from the backend, and it is async (returns true, which means I'll do).

Comment (merlin-northern) [reply client/mender/auth.go:36]: maybe lets think of a better name for FetchHWT, it seems it is more of a "Peek". not critical.

File: .gitlab-ci.yml (hunk 1 above)
Comment (tranchitella) [reply .gitlab-ci.yml:32]: Lluis confirmed this is the way to go

File: client/mender/auth.go (hunk 2 above)
Comment (tranchitella) [reply client/mender/auth.go:45]: This is the DBus API @kacf designed

Comment (merlin-northern) [reply client/mender/auth.go:45]: ah, ok then lets merge this one then.

**Review by merlin-northern — APPROVED** [review]

**Pull request merged**